### PR TITLE
Improve many things

### PR DESCRIPTION
**Pull request by ice1000**
+ Syntax change: `[| i |] A { bla }`
+ Let failure tests use `common` lib
+ Use eta expanded paths in library

### CODE REVIEW

File: base/src/main/java/org/aya/tyck/order/AyaSccTycker.java [hunk 1]
```diff
@@ -151,7 +151,7 @@ private void checkSimpleFn(@NotNull TyckOrder order, @NotNull TeleDecl.FnDecl fn
       reporter.report(new BadRecursion(fn.sourcePos, fn.ref, null));
       throw new SCCTyckingFailed(ImmutableSeq.of(order));
     }
-    decideTyckResult(fn, fn, tycker.simpleFn(newExprTycker(), fn));
+    decideTyckResult(fn, fn, tycker.simpleFn(reuseTopLevel(fn), fn));
```
Comment (imkiva) [inline base/src/main/java/org/aya/tyck/order/AyaSccTycker.java:154]: Thanks!!

**Review by imkiva — APPROVED** [review]

**Comment by ice1000** [comment]
bors merge

**Pull request merged**